### PR TITLE
Initial proposal for an openstack deployment model

### PR DESCRIPTION
**Pull request by IgnacioPro**
An initial draft on what I got so far for an OpenStack deployment model.

### CODE REVIEW

File: cibyl/models/openstack/deployment.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+
+class Deployment(Model):
+    """Openstack deployment model"""
+
+    API = { 
+        'release': {
+            'attr_type': float,
+            'arguments': [Argument(name='--release', arg_type=float,
+                description="Deployment release number")]
```
Comment (bregman-arie) [inline cibyl/models/openstack/deployment.py:29]: "Deployment release version"

File: cibyl/models/openstack/deployment.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        },                                                                                        
+        'infra_type': {
+            'attr_type': str,
+            'arguments': [Argument(name='--infra-type', arg_type=str,
+                description="Infra type")]
+        },
+        'nodes': {
+            'attr_type': str,
+            'attribute_value_class': AttributeListValue,
+            'arguments': [Argument(name='--nodes', arg_type=str,
```
Comment (bregman-arie) [inline cibyl/models/openstack/deployment.py:39]: arg_type=Node

You'll probably need to add an import like that: from cibyl.models.openstack.node import Node

File: cibyl/models/openstack/deployment.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                description="Nodes on the deployment")]
+        },
+        'service': {
+            'attr_type': Service,  
+
+            # Not sure if I should go with building another
```
Comment (bregman-arie) [inline cibyl/models/openstack/deployment.py:45]: You should add Service model (and Node model). but you can keep the mode itself empty and write "will be implemented in future PR"

File: cibyl/models/openstack/deployment.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                description="Nodes on the deployment")]
+        },
+        'service': {
+            'attr_type': Service,  
+
+            # Not sure if I should go with building another 
+            # model for the service attribute or go with
+            # just a AttributeListValue. Thoughts?
+
+            'arguments': [Argument(name='--configuration', arg_type=str,
```
Comment (bregman-arie) [inline cibyl/models/openstack/deployment.py:49]: name='--services'

**Review by bregman-arie — CHANGES_REQUESTED** [review]
Overall really good job

File: cibyl/models/openstack/deployment.py (hunk 4 above)
Comment (bregman-arie) [reply cibyl/models/openstack/deployment.py:49]: arg_type=Service

Comment (IgnacioPro) [reply cibyl/models/openstack/deployment.py:49]: Would this not be attr_type: Service and arg_type: str?

Comment (bregman-arie) [reply cibyl/models/openstack/deployment.py:49]: Yeah, meant line 43. So ignore it :)

**Comment by bregman-arie** [comment]
Looks good. @IgnacioPro  if this is ready, please mark it as such and remove [DRAFT] from PR name

**Comment by IgnacioPro** [comment]
@bregman-arie done!

**Comment by bregman-arie** [comment]
Looks good, thank you :)

**Review by bregman-arie — APPROVED** [review]

**Pull request merged**